src/org/opendatakit/briefcase/util/TransferFromODK.java

```diff
@@ -216,12 +228,46 @@ public boolean accept(File dir, String name) {
                 String safeName = scratchInstance.getName();
                 
                 int i = 2;
+                boolean same = false;
                 while (scratchInstance.exists()) {
-                  String[] contents = scratchInstance.list();
+                  File[] contents = scratchInstance.listFiles(new FilenameFilter() {
+
+                    @Override
+                    public boolean accept(File dir, String name) {
+                      return name.endsWith(".xml");
+                    }});
                   if ( contents == null || contents.length == 0 ) break;
+                  if (contents.length == 1){
+                    String itsInstanceId = null;
+                    try {
+                      XmlManipulationUtils.FormInstanceMetadata formInstanceMetadata =
+                              XmlManipulationUtils.getFormInstanceMetadata(XmlManipulationUtils.parseXml(contents[0])
+                                      .getRootElement());
+                      itsInstanceId = formInstanceMetadata.instanceId;
+
+                      //Check if the above ODK file(xml) instanceId is equal to this briefcase file instanceId then compare their MD5 hashes
+                      //if yes don't copy it, skip to next file
+                      if (itsInstanceId != null && instanceId.equals(itsInstanceId) &&

```

Comment:
I think the null check here should be on `instanceId`. If `itsInstanceId` is null, that doesn't matter because you don't dereference it. But if `instanceId` is null, then you get a null pointer exception when calling `equals`.